packages/transformer/src/IModelTransformer.ts

```diff
@@ -271,6 +271,12 @@ export interface IModelTransformOptions {
    * @default true
    */
   skipPropagateChangesToRootElements?: boolean;
+
+  /**
+   * Whether or not this transformation is a synchronization. This influences the behavior of @see [[IModelTransformer.process]].
+   * @default false
+   */
+  isSynchronization?: boolean;
```

Comment:
Maybe there is a reason why those options can't be passed to ctor.. checked what we are passing and.. turns out it is an empty object  
![image](https://github.com/iTwin/imodel-transformer/assets/28706674/dd460d15-1fdf-42d5-bda3-aeb9f15aff2a)

Reply:
`new IModelTransformer({processChangesOptions: { }})` could look strange, one might think that by passing an empty object you're opting out of processChanges, but really by passing { } you're telling me you want to run processChanges.

Reply:
I made a slight change, is it better worse or the same? 
<img width="722" alt="image" src="https://github.com/iTwin/imodel-transformer/assets/22119573/c05bae43-5225-48f7-b14d-23fba8f29b17">

Reply:
Okay i switched over to your suggestion

Reply:
We looked at code with @mindaugasdirg  and (don't be mad), we have new suggestion! `Lets leave isSynchronization as it was before (property in transformaer options) and remove processChanges options instead`, as
- allowing user to pass access token is a path to time out issues that you helped us solve recently on our end. 
- startChangeset is also no longer needed (@mindaugasdirg  knows more)

Then we would achieve 2 things: 
- all configurations would be set in ctor
- no need for if's that check both `isSynchronization` and `processChangeOptions arguments` 

Hope it would bring us closer to having options that would be very straight forward to understand and configure.

Reply:
I see that there were changes to add comment explaining that access token option should not be used, but my suggestion was to remove processChanges options (which would mean that accessToken option would be removed as well). Can you update me on latest decisions, and if you decided there is a reason to keep accessToken and startChangeset options? Thanks!

Reply:
![image](https://github.com/user-attachments/assets/9510eaeb-df5f-4f45-92fe-371ee3e86b3a)
These are the last messages I sent regarding my concerns on removing 'startChangeset' option.

Reply:
You are right, startChangeset option might be useful in some scenarios, so it makes sense to leave it. 

Suggestions for current implementation:

- Consider maybe 'ignoreMissingChangesetsInSynchronizations' could be dropped. It seems that if startChangesetIndex is set , then it already indicates that users intention is to ignore / skip missing change sets.  If you do not agree that this option can be removed, then it should be moved to `ProcessChangesOptions`.
- `unsafeFallbackSyncVersion` and `unsafeFallbackReverseSyncVersion` should be moved to `ProcessChangesOptions`
- I know you added comment to indicate that setting `accessToken` option is not safe, but maybe better approach would be to remove that option.

These suggestions are aimed to clean up and group all process changes options in one place.

Reply:
Do you think the transformer should error out explicitly if accesstoken is removed and no authorizationClient is defined on IModelHost?

Reply:
- It seems to me the same as when "access token is not passed and authorization client is not defined" in current implementation.  Do you know what error user would get right now? If error is not intuitive or clear, then it might be a good idea to check auth client configuration and provide users with better error. On the other hand, I expected `IModelHost.hubAccess.xxx()` to do the check, but it seems that it will just try to execute request with empty token (I might have missed something, haven't tried to actually execute it, only did a quick code check). 
-  Is AccessToken/Authorization client always required, or there are cases when transformation does not use access token?

Reply:
@JulijaRamoskiene I'll be working with Nick. Can you please review my new changes to address the previous comments from above?  I have moved the appropriate options from IModelTransformoptions to processChangesOptions, log warning if there is no authorization client and remove accessToken as it can be generated if there is an authorization client provided.